SRM Glaive-Class Hunter Ship

Latentish opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
The SRM has no unique ship at the moment. This adds a ship to this unrepresented PMC faction that is extremely lore accurate.
![2022 08 30-16 03 39](https://user-images.githubusercontent.com/55075747/187559198-0c832d2a-1f5a-449d-bfd0-04c0aa06cd1c.png)
![glaive](https://user-images.githubusercontent.com/55075747/187559204-59871080-1df3-41e2-85ba-2d713ffd6a7c.png)


This ship is a love letter to the Saint-Roumain Militia. This ship features: heavy influence to invest in tribal crafting, late game gun manufacturing research nodes, a glade of dirt from the homeworld of (insert name of the hunter who bought the ship here), animals and a smidge of raw meat to encourage their usual (religiously inspired) meal habits of raw or charred meats, very basic farming to reinforce tribal crafting and tribal medicine as well as feeding those who cannot eat meat, semi-decently equipped mining, and a religious hierarchy. I think this ship will have a similar effect on players to the Lamia, in that it could encourage people to actually roleplay the culture through heavy theming.
![image](https://user-images.githubusercontent.com/55075747/180596696-664a358b-d0e6-4108-b856-7150e4682e9c.png)
![image](https://user-images.githubusercontent.com/55075747/180597080-f37002c8-7bdc-4217-9f9b-b5870b8bffab.png)
![image](https://user-images.githubusercontent.com/55075747/180597094-28523cfb-8e8f-4f8d-a265-d20a70532e2e.png)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The SRM are very lore rich, are interesting, and have a culture that would be fun to emulate. This ship is a huge cultural piece that reads easily to anyone who joins it. It's very easy to see what the SRM is all about just by looking around. This is the only ship for the SRM that is not a reskin of another ship.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:phoaly
add: Glaive-class SRM ship added
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Latentish committed:
Baisic shape

-Adds the basic ship

Latentish committed:
Merge branch 'master' into SRM-Hunters-Glade

Latentish committed:
Adds most of the ship?

-Yeah

Latentish committed:
Adds more of the ship

-Fucked up wires!
-No sinks!
-Tribal medicine!
- No lightbulbs!
- No cooked meals!
- Gun!
-Religion!

Latentish committed:
A lot of stuff

- Adds 3 new nodes to research, all to do with ballistics making for the SRM
- Adds a weakened version of the sawcleaver
- Mostly finishes the ship

Latentish committed:
Makes it fly

-Makes the ship purchaseable

Latentish committed:
Small patch

- Fixes piping
- Fixes door names
- Fixes plasma engine setups
- Fixing wiring.

Latentish committed:
Another patch

- Changes the Montagnes from a chap subtype to a cap subtype
- Fixes the rigger b montagnes
- Fixes an error in the circuitboard file
- Replaces some broken subtypes on the ship
- Adds a mining locker

Latentish committed:
Another patch of note

- Fixes outfit for the montagnes in the glaive json
- Shortens the name
- Fixes SMES wiring
- Fixes some lighting options

Latentish committed:
The big update

-Adds a sermon to every roumain personnel locker.
- Fixes lighting issues
- Actually fixes the roumain montagne role by giving it a proper job subtype as chaplain
- adjusts the outfit datums in both the glaive and the riggerb

Latentish committed:
teensy patch

-Adds sermon and proper ammo to the medic's locker

Latentish commented:
![image](https://user-images.githubusercontent.com/55075747/180597427-9908528c-e7b7-44a8-9353-e95e4780065d.png)
bloodborne references

RuneQuester commented:
> ![image](https://user-images.githubusercontent.com/55075747/180597427-9908528c-e7b7-44a8-9353-e95e4780065d.png) bloodborne references

Oh dear, what was it? The lag, the griefers, or the horrible maps?
Oh, it doesn't matter. It always comes down to the admins to clean up after these sorts of messes

Latentish committed:
small tweak

-Adjust some areas
-Adds a roumain specific area
- Removes the ship from purchase

Latentish commented:
This should be test merged when the event to introduce the SRM is ready. Once that happens, I'll make the ship purchaseable, and we can get it actually tested enough to merge. So DNM for now.

Latentish committed:
Post-Test Changes

- Adds a wrench to the surgery tools
- Adds a new firstaid kit more befitting the SRM doc
- Some minor mapping changes

Latentish commented:
> This PR has been inactive for long enough to be automatically marked as stale. This means it is at risk of being auto closed in ~ 14 days, please address any outstanding review items and ensure your PR is finished, if these are all true and you are auto-staled anyway, you need to actively ask maintainers if your PR will be merged. Once you have done any of the previous actions then you should request a maintainer remove the stale label on your PR, to reset the stale timer. If you feel no maintainer will respond in that time, you may wish to close this PR youself, while you seek maintainer comment, as you will then be able to reopen the PR yourself

IT IS LITERALLY TM'D 
Die

Latentish committed:
Post Test Patch 2

- Adds the Mosin Nagant to the specialty weapons node
- Adds a proper rnd console circuit to the ship
- Adds an outpost console to the ship
-Makes the ship purchaseable

Latentish committed:
Merge branch 'master' into SRM-Hunters-Glade

Latentish committed:
Fixes Map compile error

-Updates decals to newer subtype

Latentish committed:
Post-test Patch 3

-Adds an SRM lore book to each locker on the SRM.
-Redoes the bathroom
-Replaces the oxygen floor tile with a gas tank.

Latentish committed:
This can't work

If this fixes the integration tests then I'm pissed.
I think it's angry that I put a portable pump ontop of a floor pump.

Latentish commented:
I'm FUCKING PISSED.

ArdentSpark commented:
Haha oh my god that reason for the CI fail is amusing. I had a suggestion, may I propose that these airalarms and APC's bordering the 'secret rooms' are moved? That way the option of opening up the walls and expanded into from the locker room and medical room is there, without having to move machinery
![](https://cdn.discordapp.com/attachments/673107663709798411/1008821608716181555/unknown.png)

ArdentSpark commented:
The Winchester Mk2 Rifle added by this PR to the techweb is unprintable, due to requiring Wood, which is not able to be added as a material to Circuit printers, Protolathes, Autolathes, ORMs or Silos due to not containing sufficient materials
![](https://cdn.discordapp.com/attachments/673107663709798411/1008859493012099233/unknown.png)
![](https://cdn.discordapp.com/attachments/673107663709798411/1008859517552959539/unknown.png)

Latentish commented:
I'll finish this and mark it ready for review later

Latentish committed:
Merge remote-tracking branch 'upstream/master' into SRM-Hunters-Glade

Latentish committed:
Glaive-Patch-something

-Moved some machines to make expansion easier.
-Replaced tinyfans with a shieldgen
- Removed wood from the winchester recipe
- Fixed all instances of "whitesands" objects

PrefabQuasar reviewed (changes requested):
this map doesnt seem to have any major issues, nor have any shown up when it was TM'd, however, i'd reccomend either changing the wardrobe lockers to another sprite or making them not access locked, as the sprite currently used isnt designed to have an access lock, so it looks odd on them, i'm also unsure on the inclusion of sentience potions in here, i feel like too often those can just lead to "animal that acts like a human in every respect", which doesnt feel great for RP

PrefabQuasar commented:
> The Winchester Mk2 Rifle added by this PR to the techweb is unprintable, due to requiring Wood, which is not able to be added as a material to Circuit printers, Protolathes, Autolathes, ORMs or Silos due to not containing sufficient materials ![](https://camo.githubusercontent.com/0d279ec6bb0798a8ce0590f992302b29945db9866156350e4a57472f152cfead/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3637333130373636333730393739383431312f313030383835393439333031323039393233332f756e6b6e6f776e2e706e67) ![](https://camo.githubusercontent.com/96a62b22d8b98afa6c2948f2f81a375c85874bd7d756b11c45b54b4af95d5eec/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3637333130373636333730393739383431312f313030383835393531373535323935393533392f756e6b6e6f776e2e706e67)

assuming that this has been addressed

Latentish commented:
> assuming that this has been addressed

Yes this has been addressed and the wood has been removed from the requirement.

Latentish commented:
> this map doesnt seem to have any major issues, nor have any shown up when it was TM'd, however, i'd reccomend either changing the wardrobe lockers to another sprite or making them not access locked, as the sprite currently used isnt designed to have an access lock, so it looks odd on them, i'm also unsure on the inclusion of sentience potions in here, i feel like too often those can just lead to "animal that acts like a human in every respect", which doesnt feel great for RP

I can swap the locker sprite.

The sentience potion here is for part of the ship's niche of beast taming. I get that players might act like normal people as sentient animals, but I feel that's a reflection on sentience potions as a whole. I am unsure if there's another potion to let players take over simples.

PrefabQuasar commented:
> > this map doesnt seem to have any major issues, nor have any shown up when it was TM'd, however, i'd reccomend either changing the wardrobe lockers to another sprite or making them not access locked, as the sprite currently used isnt designed to have an access lock, so it looks odd on them, i'm also unsure on the inclusion of sentience potions in here, i feel like too often those can just lead to "animal that acts like a human in every respect", which doesnt feel great for RP
> 
> I can swap the locker sprite.
> 
> The sentience potion here is for part of the ship's niche of beast taming. I get that players might act like normal people as sentient animals, but I feel that's a reflection on sentience potions as a whole. I am unsure if there's another potion to let players take over simples.

i'm still iffy on them, i'd want a third opinion really

Apogee-dev commented:
> i'm still iffy on them, i'd want a third opinion really

Sentience potions are fine imo due to mechanical need. The beast tamer gimmick just plain doesn't work without them, and it gives ghosts more stuff to do.

PrefabQuasar commented:
good enough for me, although one day i'd hope to see a potion that gives sentience but not the common language. actually that might not be too hard to add

PrefabQuasar reviewed (approved):
regardless, thats everything i had adressed, and nothing's come up during testing

Latentish committed:
Switches out the closets

- Requested change of swapping the closet sprites. Changed them to the cabinet.

Latentish committed:
Why the fuck is this here

Reverts a change I didn't make but probably happened because I opened the lux at some point.

Fikou commented on code/game/MapData/shuttles/srm_glaive.dm:
```diff
@@ -0,0 +1,176 @@
+		/obj/item/reagent_containers/food/snacks/grown/ash_flora/cactus_fruit = 3,
+		/obj/item/reagent_containers/food/snacks/meat/slab/bear = 3,
+		/obj/item/reagent_containers/food/snacks/grown/ash_flora/mushroom_leaf = 3)
+	generate_items_inside(items_inside,src)
```
```suggestion
	var/static/list/items_inside = list(
		/obj/item/healthanalyzer = 1,
		/obj/item/reagent_containers/food/snacks/grown/ash_flora/puce = 1,
		/obj/item/reagent_containers/glass/mortar = 1,
		/obj/item/reagent_containers/glass/bowl/mushroom_bowl = 1,
		/obj/item/pestle = 1,
		/obj/item/reagent_containers/food/snacks/grown/ash_flora/cactus_fruit = 3,
		/obj/item/reagent_containers/food/snacks/meat/slab/bear = 3,
		/obj/item/reagent_containers/food/snacks/grown/ash_flora/mushroom_leaf = 3,
	)
	generate_items_inside(items_inside, src)
```

Fikou committed:
Update code/game/MapData/shuttles/srm_glaive.dm
